tests/zkasm/run-tests-zkasm.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable no-use-before-define */
+const path = require('path');
+const fs = require('fs');
+const chalk = require('chalk');
+const zkasm = require('@0xpolygonhermez/zkasmcom');
+const smMain = require('@0xpolygonhermez/zkevm-proverjs/src/sm/sm_main/sm_main');
+const {
+    compile,
+    newCommitPolsArray
+} = require('pilcom');
+const buildPoseidon = require('@0xpolygonhermez/zkevm-commonjs').getPoseidon;
+
+const emptyInput = require('@0xpolygonhermez/zkevm-proverjs/test/inputs/empty_input.json');
+
+const {
+    argv
+} = require('yargs')
+    .alias('v', 'verbose');
+
+// Global paths to build Main PIL to fill polynomials in tests
+const pathMainPil = path.join(__dirname, 'node_modules/@0xpolygonhermez/zkevm-proverjs/pil/main.pil');
+const fileCachePil = path.join(__dirname, 'node_modules/@0xpolygonhermez/zkevm-proverjs/cache-main-pil.json');
+
+async function main() {
+    // Compile pil
+    console.log(chalk.yellow('--> Compile PIL'));
+    const cmPols = await compilePil();
+
+    // Get all zkasm files
+    const pathZkasm = path.join(process.cwd(), process.argv[2]);
+    const files = await getTestFiles(pathZkasm);
+
+    let hasUnexpectedFailures = false;
+    // Run all zkasm files
+    // eslint-disable-next-line no-restricted-syntax
+    console.log(chalk.yellow('--> Start running zkasm files'));
+    for (const file of files) {
+        if (file.includes('ignore'))
+            continue;
+
+        let shouldFail = file.split("/").pop().startsWith("_should_fail_");
+        let testFailed = await runTest(file, cmPols);
+        hasUnexpectedFailures |= (testFailed && !shouldFail) || (shouldFail && !testFailed);
+    }
+    if (hasUnexpectedFailures) {
+        process.exit(1);
+    }
+}
+
+async function compilePil() {
+    if (!fs.existsSync(fileCachePil)) {
+        const poseidon = await buildPoseidon();
+        const {
+            F
+        } = poseidon;
+        const pilConfig = {
+            defines: {
+                N: 4096
```

Comment:
What is this define?

Reply:
As far as I can tell, it is a number of rows, allocated for Main state machine [1].

I'm not quite sure what is the effect of having a different value here and in ROM config below in the file. I'll raise a question with Polygon folks.

For now, bumped the number to 2**23 to match the one below, as it doesn't seem to influence the execution time and seems like a safer solution.

[1] https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/main.pil#L13

Reply:
Ok, looks like based on the answer on TG, for testing, we can keep the number of rows for Main low.